boards/espressif/esp32_devkitc_wroom/Kconfig.esp32_devkitc_wroom

```diff
@@ -3,3 +3,5 @@
 
 config BOARD_ESP32_DEVKITC_WROOM
 	select SOC_ESP32_WROOM_32UE_N4
+	select SOC_ESP32_PROCPU if BOARD_ESP32_DEVKITC_WROOM_ESP32_PROCPU
+	select SOC_ESP32_APPCPU if BOARD_ESP32_DEVKITC_WROOM_ESP32_APPCPU
```

Comment:
@nordicjm moving this SOC definition into here was my initial approach. However, you suggested me to move this into `board/Kconfig` instead. Any comments?

Reply:
@tejlmand Anyway, this move won't work. Shall I restore to `kconfig` instead? ( I mean, keep in there):
```c
error: SOC_ESP32_PROCPU (defined at soc/espressif/esp32/Kconfig:20) is assigned in a configuration
file, but is not directly user-configurable (has no prompt). It gets its value indirectly from other
symbols.
```

Reply:
If you are selecting an actual core then yes it belongs here, but that was not the case in the original PR, these selected something relating to an AMP configuration

Reply:
@tejlmand, `SOC_ESP32_PROCPU` does not select a SoC. It only tells the build system which core is the build intended for, thus allowing proper SoC initialization (one or the other). 
1. Should we rename this and remove the `SOC_` prefix?
2. If we keep, should this be kept where it is instead moving to `Kconfig.soc`?

Reply:
it seems like two symbols might be needed.
I'm not into the details on how the current `SOC_ESP32_PROCPU` / `SOC_ESP32_APPCPU`  symbols are used, but this thread indicates that it's different than a CPU cluster symbol reflecting the `soc.yml` setting.

I'll leave it to you to decide if the symbols can be united to a single one, or if two independent symbols are required.

But we should have a symbol which can be selected by the board to indicate the CPU Cluster for which the build is targeted. Especially for use-case like the ones in this PR.